detect_secrets_server/storage/core/git.py

```diff
@@ -149,17 +157,16 @@ def _filter_filenames_from_diff(directory, last_commit_hash):
     ]
 
 
-def _git(directory, *args):
+def _git(directory, *args, **kwargs):
     output = subprocess.check_output(
         [
             'git',
             '--git-dir', directory,
-
-            # Work-tree is required for some git commands, because of bare repos.
-            # However, it doesn't hurt to put it for all of them.
-            '--work-tree', '.',
         ] + list(args),
         stderr=subprocess.STDOUT
-    )
+    ).decode('utf-8', errors='ignore')
 
-    return output.decode('utf-8', 'ignore').strip()
+    # This is to fix https://github.com/matiasb/python-unidiff/issues/54
+    if 'dont_strip_output' in kwargs:
```

Comment:
Let's avoid the use of double negatives, because that's unnecessary complication.

```
def _git(directory, *args, **kwargs):
    kwargs['should_strip_output'] = kwargs.get('should_strip_output', True)
    ...
    if not kwargs['should_strip_output']:
        return output

    return output.strip()
```

Reply:
++, initially had it as `strip_output=True` as a default arg, but Python 2, your suggestion is much improved until we deprecate Python 2.

(As an aside, I'm in favor of deprecating Python 2 for -server, I can make the GitHub issue if you agree, as the pre-commit has to have legacy support, but not the server.)